sopel/modules/clock.py

```diff
@@ -54,14 +54,29 @@ def setup(bot):
 
 @module.commands('t', 'time')
 @module.example('.t America/New_York')
+@module.example('.t Exirel')
+@module.example('.t #sopel')
 def f_time(bot, trigger):
-    """Returns the current time."""
+    """Returns the current time.
+
+    The command takes an optional parameter: it can be either a timezone, a
+    nick, or a channel.
+    """
     if trigger.group(2):
-        zone_name = trigger.group(2).strip()
+        argument = trigger.group(2).strip()
+        channel_or_nick = tools.Identifier(argument)
+
         try:
-            zone = validate_timezone(zone_name)
+            if channel_or_nick in bot.users:
+                zone = get_timezone(bot.db, bot.config, None, channel_or_nick,
+                                    trigger.sender)
```

Comment:
This seems dangerous. Falling back on the current channel's timezone could give misleading results if the given nick isn't in the current channel. Hell, they could be misleading if the given nick *is* in the current channel.

Actually, falling back on anything but the user's own timezone setting is probably not a good idea. I think this should emit an error if the user hasn't set a timezone, rather than give an unclear result that may or may not be in the user's own timezone.

Reply:
Well, it's kind of what the previous code did, with less success I believe. I think I went with a conservative approach here: I kept the same behavior as today.

I'd suggest to do another PR to raise more errors, because that means changing the behavior of (multiple) command(s).

Reply:
I'm having second thought on that one.

Let's raise an error.